3-UsingAPIs/Week2/homework/ex2-pokemonApp/index.js

```diff
@@ -24,33 +24,35 @@ parameters and return values to pass data back and forth.
 ------------------------------------------------------------------------------*/
 async function fetchData(url) {
   const response = await fetch(url);
-  const data = await response.json();
-
-  return new Promise((resolve, reject) => {
-    if (response.ok) {
-      try {
-        resolve(data);
-      } catch (error) {
-        reject(error.message);
-      }
-    }
-  });
+  if (response.ok) {
+    return response.json();
+  }
+  throw new Error('HTTP ERROR');
 }
 
-function fetchAndPopulatePokemons(name) {
-  const option = document.createElement('option');
-  option.textContent = name;
-  option.classList = 'pokemon-options';
+async function fetchAndPopulatePokemons() {
   const selectOption = document.getElementsByTagName('select')[0];
-  selectOption.appendChild(option);
+
+  const data = await fetchData(`https://pokeapi.co/api/v2/pokemon?limit=151`);
+
+  data.results.forEach((pokemon) => {
+    console.log(pokemon);
+    const option = document.createElement('option');
+    option.textContent = pokemon.name;
+    option.setAttribute('value', pokemon.url);
```

Comment:
Excellent to use the `.value` attribute to hold the `url` of the image. Note that you can assign it directly without using `setAttribute()`:

```suggestion
    option.value = pokemon.url;
```